DiscordCardLinker/CardBot.cs

```diff
@@ -311,166 +318,123 @@ public class CardBot {
-			if(Cache.Any(x => x.searchString == searchString)) {
-				var card = Cache.Where(x => x.searchString == lowerSearch).First().card;
-				candidates.Add(card);
-				return candidates;
-			}
-
-			if (CardSubtitles.ContainsKey(lowerSearch)) {
-				candidates.AddRange(CardSubtitles[lowerSearch]);
-				return candidates;
-			}
+			string lowerSearch = ScrubInput(searchString);
 
 			if (CardCollInfo.ContainsKey(lowerSearch)) {
 				candidates.Add(CardCollInfo[lowerSearch]);
-				return candidates;
 			}
 
 			if (CardFullTitles.ContainsKey(lowerSearch)) {
 				candidates.AddRange(CardFullTitles[lowerSearch]);
-				return candidates;
 			}
 
 			if (CardTitles.ContainsKey(lowerSearch)) {
 				candidates.AddRange(CardTitles[lowerSearch]);
-				return candidates;
+			}
+
+			if (CardSubtitles.ContainsKey(lowerSearch))
+			{
+				candidates.AddRange(CardSubtitles[lowerSearch]);
 			}
 
 			if (CardNicknames.ContainsKey(lowerSearch)) {
 				candidates.AddRange(CardNicknames[lowerSearch]);
-				return candidates;
 			}
 
-			//TODO: fuzzy search on all of the above
+			if (lowerSearch.Length > 2)
+			{
+				foreach (var key in CardFullTitles.Keys)
+				{
+					if (key.Contains(lowerSearch))
+					{
+						candidates.AddRange(CardFullTitles[key]);
+					}
+				};
+			}
 			return candidates;
 		}
 
-		private async Task SendSingle(MessageCreateEventArgs e, CardDefinition card, MatchType type, string search) {
-			switch (type) {
-				case MatchType.Image:
-					await SendImage(e, card);
-					break;
-				case MatchType.Wiki:
-					await SendWikiLink(e, card);
-					break;
-			}
+		private DiscordButtonComponent DeleteButton(ulong AuthorID)
+		{
+			return new DiscordButtonComponent(ButtonStyle.Danger, $"delete_{AuthorID}", "Delete");
+		}
 
-			AddSuccessfulSearch(search, card);
+		private DiscordButtonComponent LockinButton(ulong AuthorID, bool disabled = false)
+		{
+			return new DiscordButtonComponent(ButtonStyle.Primary, $"lockin_{AuthorID}", "Accept", disabled);
 		}
 
-		private async Task SendImage(MessageCreateEventArgs e, CardDefinition card) {
-			await e.Message.RespondAsync(card.ImageURL);
+		private async Task SendImage(DiscordMessage original, CardDefinition card)
+		{
+			await original.RespondAsync(BuildSingle(original, card, true, true, false));
 		}
 
-		private async Task SendWikiLink(MessageCreateEventArgs e, CardDefinition card) {
-			await e.Message.RespondAsync("Information about " + card.Title + " is available on scomp at: " + card.WikiURL);
-			//var msg = await new DiscordMessageBuilder()
-			//	.With
-			//e.Message.RespondAsync()
+		private DiscordMessageBuilder BuildSingle(DiscordMessage original, CardDefinition card, bool wiki, bool buttons, bool disable)
+		{
+			var builder = new DiscordMessageBuilder()
+				.WithReply(original.Id)
+				.WithContent(card.ImageURL);
+
+			var comps = new List<DiscordComponent>();
+
+			if (wiki)
+			{
+				comps.Add(new DiscordLinkButtonComponent(card.WikiURL, "Wiki", false));
```

Comment:
SWCCG does not have a "Wiki". This button should say "scomp"

Reply:
Fair enough.